massa-async-pool/src/bootstrap.rs

```diff
@@ -0,0 +1,58 @@
+//! Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+//! Provides serializable strucutres for bootstrapping the AsyncPool
+
+use crate::message::AsyncMessage;
+use massa_models::{
+    DeserializeCompact, DeserializeVarInt, ModelsError, SerializeCompact, SerializeVarInt,
+};
+use serde::{Deserialize, Serialize};
+
+/// Represents a snapshot of the async pool aimed at bootstrapping an AsyncPool
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AsyncPoolBootstrap {
+    /// list of async messages in no particular order
+    pub(crate) messages: Vec<AsyncMessage>,
+}
```

Comment:
I don't think it's a good practice to always turn struct with one field to a new type pattern since when accessing the field you make it by 0 instead of the real name which loose clarity.

Reply:
As it's a vec, it seems weird to wrap it in a struct. Maybe a new type pattern
```suggestion
/// list of async messages in no particular order
pub type AsyncPoolBootstrap = Vec<AsyncMessage>;
```

Reply:
To keep harmonization with ledger and be able to have more data in the future it's also better to have a structure. Also when you use it you will directly use the vec methods which is not clear and will need changes all accross the code if we need an other field.

Reply:
If we need another field? Like what?

If that struct evolves soon, ok, we would like a struct. But if it's just to hide a data structure, a type is enough. As `AsyncMessage` is in our crate, we also don't care about the orphan rule, and we can impl whatever we want as if it was a struct.

Reply:
I don't think it's for hide a type as it's a structure for Bootstrap (as I deduce from the name) if in the future we want to for example save a value for the max number of messages etc...

Reply:
Since this debate won't end due to it being related to personal preference, I will decide "arbitrarily" to unlock the situation, but we can discuss this afterwards. 

=> let's keep it as is for now